src/MicrosoftTeams.ts

```diff
@@ -421,22 +422,61 @@ namespace microsoftTeams
             let link = document.createElement("a");
             link.href = authenticateParameters.url;
 
-            let messageId = sendMessage(
-                "authentication.authenticate",
-                link.href,
-                authenticateParameters.width,
-                authenticateParameters.height);
-            callbacks[messageId] = (success: boolean, response: string) =>
+            // Clear any previous success/failure values
+            localStorage.removeItem("authentication.success");
+            localStorage.removeItem("authentication.failure");
+
```

Comment:
Yes, the auth completion endpoint will have to be in the same domain as the endpoint that kicked off the auth flow. I will make sure to document this before asking developers to switch to it. At this point I don't know of a way to make this work across domains without message posting which depends on window.opener. Once auth completes the tab content/config page would be able to use navigateCrossDomain to move to a different domain if it wants (and that's what VSTS does today).